app/preprints/-components/submit/preprint-state-machine/component.ts

```diff
@@ -68,11 +69,17 @@ export default class PreprintStateMachine extends Component<StateMachineArgs>{
             this.preprint = this.args.preprint;
             this.setValidationForEditFlow();
             this.isEditFlow = true;
+            if (this.args.preprint.reviewsState === ReviewsState.REJECTED) {
+                this.displayFileUploadStep = true;
+            } else {
+                this.displayFileUploadStep = false;
+            }
```

Comment:
When this component is invoked in the "Edit an existing preprint" workflow (`this.args.preprint` being truthy in line 68), this bit of logic will be triggered. If the preprint that is passed in has `reviews_state: 'rejected'`, the File upload step will be shown, otherwise the File upload step will be skipped